test/e2e/runner.go

```diff
@@ -51,6 +51,11 @@ func main() {
 
 	// Only provision a cluster if there isnt a name present
 	if cfg.Name == "" {
+		if cfg.SoakClusterName != "" {
```

Comment:
Should this belong to some cleanup step (maybe the teardown function at the bottom of this file)? The comments on this section of code says it is for provisioning a cluster.

Reply:
The reason why I put this here is that when generating a new soak cluster it is necessary to delete the previously existing soak cluster before deploying a new one since they use the same resource group (and we want to make sure we don't keep them after we're done testing on them).

Reply:
I see, thanks for the explanation.